nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/repository/FileSystemRepository.java

```diff
@@ -149,6 +154,10 @@ public FileSystemRepository(final NiFiProperties nifiProperties) throws IOExcept
         for (final Path path : fileRespositoryPaths.values()) {
             Files.createDirectories(path);
         }
+        this.maxFlowFilesPerClaim = nifiProperties.getMaxFlowFilesPerClaim();
+        this.writableClaimQueue  = new LinkedBlockingQueue<>(maxFlowFilesPerClaim);
+        final String maxAppendableClaimSize = nifiProperties.getMaxAppendableClaimSize();
+        this.maxAppendableClaimLength = DataUnit.parseDataSize(maxAppendableClaimSize, DataUnit.B).intValue();
```

Comment:
If this value gets set to something like "10 GB" this could cause some really problematic (and difficult to track down) problems because the value would overflow to a negative value. Probably is best to use a longValue() and then perhaps even cap it at something like 100 MB or 10 MB and if the value is larger than that, just emit a WARN log event and use the max value.